src/wp-includes/rest-api.php

```diff
@@ -2109,13 +2393,36 @@ function rest_parse_embed_param( $embed ) {
  *
  * @since 5.5.0
  * @since 5.6.0 Support the "patternProperties" keyword for objects.
+ *              Support the "anyOf" and "oneOf" keywords.
  *
  * @param array|object $data    The response data to modify.
  * @param array        $schema  The schema for the endpoint used to filter the response.
  * @param string       $context The requested context.
  * @return array|object The filtered response data.
  */
 function rest_filter_response_by_context( $data, $schema, $context ) {
+	if ( isset( $schema['anyOf'] ) ) {
+		$matching_schema = rest_find_any_matching_schema( $data, $schema, '' );
+		if ( ! is_wp_error( $matching_schema ) ) {
+			if ( ! isset( $schema['type'] ) ) {
+				$schema['type'] = $matching_schema['type'];
+			}
+
+			$data = rest_filter_response_by_context( $data, $matching_schema, $context );
```

Comment:
Should we be returning here? Do we need to go thru the checks again?

Reply:
Filtration here uses only the matching schema. After that, we should continue working with the modified data to ensure that the main schema's keywords are also processed.

Reply:
So this would be if the `anyOf` schemas don't have any `type` information attached?

Reply:
We have four possible scenarios here:
- both parent and child schemas have `type` - we don't do anything in this case
- parent schema has `type,` but child schema doesn't - we set parents `type` to the child before matching (see `rest_find_any_matching_schema` and `rest_find_one_matching_schema`)
- parent schema doesn't have `type,` but child schema do - we set `type` from the matching child schema to the parent schema to avoid errors in the following processing
- both schemas don't have `type` - we don't do anything, and validation fails with error (because the current implementation requires `type`).

Is it an answer to your question?

Reply:
Understood, thanks for the clarification!